### PR TITLE
Add v4 TFTP Server Support

### PR DESCRIPTION
**Pull request by ryanheffernan**
## Adding V4 TFTP Server Support

V4 host addresses did not work due to always using AF_INET6 when binding the server socket. I was looking to use this in a environment where netbooting over v6 is not possible (yet) and it was a simple fix to add in v4 support.

### CODE REVIEW

**Comment by facebook-github-bot-7** [comment]
By analyzing the blame information on this pull request, we identified @pallotron and @berkerpeksag to be potential reviewers.

**Comment by facebook-github-bot-2** [comment]
@ryanheffernan updated the pull request.

**Comment by pallotron** [comment]
Thanks for your PR!

Uhm, IPv4 should be already supported, we use `fbtftp` in production in all of our datacenters (on both dual stacked and v6only environments) and on my development server it works as expected. 

There is a notion of "mapped" IPv4 addresses, which allows to have a single IPv6 server socket accept both IPv4 and IPv6 connections. How this precisely works depends on the operating system.
On Linux, "mapped" addresses are the default. Open a v6 socket, and connect to it through a v4 client, and it will just work. For this, you need to specify the "any" (`::`) server address.

I guess your Linux server doesn't have ipv6 enabled at all?

Excerpt from my personal devserver:

```
$ ifconfig eth0
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.166.47  netmask 255.255.255.0  broadcast 10.35.166.255
        inet6 fe80::92e2:baff:fe95:b928  prefixlen 64  scopeid 0x20<link>
        inet6 2401:db00:3011:26:face:0:1b:0  prefixlen 64  scopeid 0x0<global>
```

```
$ python3.5 examples/server.py --root /tmp/tftp --port 1069
DEBUG:root:Starting the metrics callback in 60s
INFO:root:New connection from peer `('::ffff:10.35.166.47', 61901, 0, 0)` asking for path `file.txt`
INFO:root:Options requested from peer ('::ffff:10.35.166.47', 61901, 0, 0):  {'mode': 'netascii'}
INFO:root:Options to ack for peer ('::ffff:10.35.166.47', 61901, 0, 0):  {}
INFO:root:Stats: for ('::ffff:10.35.166.47', 61901, 0, 0) requesting 'file.txt'
INFO:root:Error: {}
INFO:root:Time spent: 5ms
INFO:root:Packets sent: 1
INFO:root:Packets ACKed: 0
INFO:root:Bytes sent: 12
INFO:root:Options: {}
INFO:root:Blksize: 512
INFO:root:Retransmits: 0
INFO:root:Server port: 1069
INFO:root:Client port: 61901
DEBUG:root:Closing response data object
DEBUG:root:Closing socket
DEBUG:root:Dying.
```

```
$ tftp 10.35.166.47 1069
tftp> get file.txt
tftp>
```

**Comment by ryanheffernan** [comment]
… Derp.

I totally missed that. I simply tried to bind to a v4 address and got the socket.gaierror, so assumed v4 wasn't supported. Totally forgot about mapped addresses :). I'll pull the PR.

Thanks for the quick response!

**Comment by pallotron** [comment]
So I think there is still space for your make file change though :)

**Comment by pallotron** [comment]
I think I'd be happy to take the Makefile change if you remove the rest of the python changes from the PR :)

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,11 @@
-PYTHON=python3  # if there's no python3 binary this will fail first
+PYTHON=python3
```
Comment (pallotron) [inline Makefile]: basically only leave these lines and remove the rest :)

**Comment by ryanheffernan** [comment]
I'll close this PR and submit one dedicated to the makefle changes.

**Pull request closed without merging**